python-frontend/src/test/java/org/sonar/python/semantic/v2/ProjectLevelTypeTableTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.semantic.v2;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.sonar.python.semantic.ProjectLevelSymbolTable;
+import org.sonar.python.types.v2.ClassType;
+import org.sonar.python.types.v2.ModuleType;
+import org.sonar.python.types.v2.PythonType;
+
+class ProjectLevelTypeTableTest {
+
+  @Test
+  void getTypeTest() {
```

Comment:
Should we only focus on synthetic types here?
While the method is created with this use case in mind, this is probably not the only one for it.